File: src/main/java/core/TranscriptomePerspective.java

package core;

import java.io.File;
import java.io.IOException;

import core.utils.Genome;
import core.utils.Perspectives;
import core.utils.Transcriptome;

/**
 * The tarnscriptome perspective on a clone.
 * @author noemi
 *
 */
public class TranscriptomePerspective extends Perspective{

	public TranscriptomePerspective(float f, String sampleName, String[] nMut) {
		super(f, sampleName);
		this.profile=new Transcriptome(nMut);
	}

	public TranscriptomePerspective(File outFile, String rootName) throws Exception {
		super(outFile,Perspectives.TranscriptomePerspective,rootName);
	}

	@Override
	public Clone getPerspective(Perspectives whichP) {
		if(whichP.equals(Perspectives.valueOf(this.getClass().getSimpleName()))){
			return this;
		}
		return null;
	}

}

File: webportal/module1/src/lib/cloneid/core/TranscriptomePerspective.js
import { Perspective } from './Perspective';
import { Perspectives } from './utils/Perspectives';
import { Transcriptome } from './utils/Transcriptome';

export class TranscriptomePerspective extends Perspective {
  constructor(f, sampleName, nMut) {
    super(f, sampleName);
    this.profile = new Transcriptome(nMut);
  }


  getPerspective(whichP) {
    if (whichP === Perspectives[this.constructor.name]) {
      return this;
    }
    return null;
  }
}